odxtools/diagcodedtype.py

```diff
@@ -43,9 +61,10 @@ def get_static_bit_length(self) -> Optional[int]:
         return None
 
     @property
-    @abc.abstractmethod
     def dct_type(self) -> DctType:
-        pass
+        odxraise(f"Class {type(self).__name__} does not override required method "
+                 f"dct_type()", NotImplementedError)
+        return "STANDARD-LENGTH-TYPE"
```

Comment:
why is this kind of weird abc implementation?

Reply:
the problem with normal ABCs is that they cannot be instantiated, yet we need to do so for the `from_et()` pattern to work with class hierarchies. the `return "STANDARD-LENGTH-TYPE"` statement is added to have a fighting chance of the calling code not crashing in non-strict mode.

Reply:
what error are you getting?

Reply:
I have not seen any, but this code path is taken in non-strict mode if an unimplemented DCT is encountered, and the normal procedure for non-strict mode is to try to continue gracefully.

Reply:
those kind of problems are solved by making a NotSupprtedDtc class, instead of creating an instance of an abstract class

Reply:
how do you do inheritance then? Instantiating `NotSupportedDtc` in `.from_et()` whilst deriving from a different class feels pretty unnatural for me for this...

(note that in C++ the constructors could simply be overloaded, but unfortunately this is impossible with python.)

Reply:
what dtc type we still don't support?

Reply:
as far a I can see we currently have all mentioned by the ODX 2.2 XSD, i.e., this code path only affects incorrect files or files using a future version of the standard (assuming that this version introduces new or renames existing DCTs).

Reply:
So you can just return `None` ?

Reply:
yeah, good point. that can be done. fixed: https://github.com/mercedes-benz/odxtools/pull/304/commits/2ddb472981b0278c3d864622e44650e9101359ba